### PR TITLE
Networking v2: Add Subnet Data Source

### LINKED ISSUE #129: Add openstack_networking_subnet_v2 datasource
T̀he `lb_loadbalancer_v2` resource requires passing `vip_subnet_id` but we don't currently have a way to get it from the API.
It would be nice if we could do that:
```hcl
data "openstack_networking_subnet_v2" "subnet" {
  name = "my-subnet"
}

resource "openstack_lb_loadbalancer_v2" "lb" {
  vip_subnet_id = "${data.openstack_networking_subnet_v2.subnet.id}"
}
```

### PR DESCRIPTION
**Pull request by jtopjian**
This commit adds the `openstack_networking_subnet_v2` data source
which is able to query for a subnet based on given parameters.

Fixes #129 

/cc @mcanevet

### CODE REVIEW

**Comment by yanndegat** [comment]
fuck;)

**Comment by jtopjian** [comment]
@yanndegat haha - crazy timing :)

I apologize - I should have made a comment in #129 that I would work on this.

Given that this data source has a few more features than the one you submitted, we can do two things:

1. If you'd like to add the missing bits to your PR, I'm happy to close this one in favor of yours
2. We'll use this PR since it was submitted first.

**Comment by yanndegat** [comment]
of course we use this PR

you're first and moreover your datasource is much more complete than mine

thanks a lot for this datasource because it's really useful

best regards

**Comment by mcanevet** [comment]
@jtopjian @yanndegat if you have spare time you can work on this one : https://github.com/terraform-providers/terraform-provider-openstack/issues/122 :-)

**Comment by jtopjian** [comment]
@mcanevet  The main problem with the `clouds.yaml` support is that it'll probably take me 2-3 evenings to do. I need to first get a generic version of it into `gophercloud/utils`. I'll fully admit to doing other smaller features first since I can get them done faster, but I promise you I will work on `clouds.yaml` before any other features next :smile:

**Comment by yanndegat** [comment]
@mcanevet well, not sure i'll find time to work on this right now, 

i've got plenty of other things to do.

may i contact you by email ? i may have some questions for you.

**Comment by mcanevet** [comment]
@yanndegat sure, you may find my email in on of my commit log.

File: openstack/data_source_openstack_networking_subnet_v2.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+	// Set the allocation_pools
+	var allocationPools []map[string]interface{}
+	for _, v := range subnet.AllocationPools {
+		pool := make(map[string]interface{})
+		pool["start"] = v.Start
+		pool["end"] = v.End
+
+		allocationPools = append(allocationPools, pool)
+	}
+	d.Set("allocation_pools", allocationPools)
```
Comment (catsby) [inline openstack/data_source_openstack_networking_subnet_v2.go]: For any non-scalar value, we should check the error returned from `d.Set()`

```
if err := d.Set("allocation_pools", allocationPools); err != nil {
  // probably just log in this scenario
}
```

Comment (catsby) [reply openstack/data_source_openstack_networking_subnet_v2.go]: The same goes for `dns_nameservers` and `host_routes`; basically any `TypeSet`, `TypeList`, or `TypeSchema`

Comment (jtopjian) [reply openstack/data_source_openstack_networking_subnet_v2.go]: @catsby Thanks for pointing this out - I'll get this cleaned up.

Is this fairly new? This isn't done anywhere else in the OpenStack provider (which I can look at fixing).

Comment (catsby) [reply openstack/data_source_openstack_networking_subnet_v2.go]: Nope, not new at all, pretty old actually 🤔 

Unfortunately it's not as enforced as it should be, so community members aren't 100% aware of things like this. This is a Terraform Team failure 😦 We need a best-practices guide!

**Comment by jtopjian** [comment]
@fatmcgav Since @catsby already did a review, I'm going to go ahead and merge this.

@catsby Thanks!

**Pull request merged**